src/main/java/mods/railcraft/common/carts/EntityTunnelBore.java

```diff
@@ -820,37 +820,34 @@ protected boolean mineBlock(BlockPos targetPos, BlockRailBase.EnumRailDirection
         if (!canMineBlock(targetPos, targetState))
             return false;
 
-        // Start of Event Fire
-        BreakEvent breakEvent = new BreakEvent(world, targetPos, targetState, CartTools.getCartOwnerEntity(this));
-        MinecraftForge.EVENT_BUS.post(breakEvent);
+        EntityPlayerMP fakePlayer = CartTools.getFakePlayerWith(this, head);
 
-        if (breakEvent.isCanceled())
+        // Fires break event within; harvest handled separately
+        if (!WorldPlugin.playerRemoveBlock(world, targetPos, fakePlayer, false))
             return false;
-        // End of Event Fire
 
-        boolean silk = EnchantmentHelper.getEnchantmentLevel(Enchantments.SILK_TOUCH, head) > 0;
-        NonNullList<ItemStack> items = NonNullList.create();
-        int fortuneLevel = EnchantmentHelper.getEnchantmentLevel(Enchantments.FORTUNE, head);
+        if (RailcraftConfig.borePreserveStacks() && world.getGameRules().getBoolean("doTileDrops")) {
+            boolean silk = EnchantmentHelper.getEnchantmentLevel(Enchantments.SILK_TOUCH, head) > 0;
+            NonNullList<ItemStack> items = NonNullList.create();
+            int fortuneLevel = EnchantmentHelper.getEnchantmentLevel(Enchantments.FORTUNE, head);
 
-        if (silk) {
-            ItemStack stack = HarvestPlugin.getSilkTouchDrop(targetState);
-            if (!InvTools.isEmpty(stack)) {
-                items.add(stack);
+            if (silk) {
+                ItemStack stack = HarvestPlugin.getSilkTouchDrop(targetState);
+                if (!InvTools.isEmpty(stack)) {
+                    items.add(stack);
+                }
+                // Use modifiable lists for events
+            } else {
+                targetState.getBlock().getDrops(items, world, targetPos, targetState, fortuneLevel);
```

Comment:
If the block is removed when this line is called, it could return the wrong drops yes? At the very least any tile entities are probably gone.

Reply:
It is the same code as the old one. https://github.com/Railcraft/Railcraft/pull/1760/files#diff-601fb497a36abaa013f7ec1a8fb43f9aL842 The indent was just changed, so you may think this is the "incorrect new code."

Reply:
Fixed.